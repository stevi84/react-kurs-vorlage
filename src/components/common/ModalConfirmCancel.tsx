import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle } from '@mui/material';

interface ModalConfirmCancelOwnProps {
  isOpen: boolean;
  headerText: string;
  bodyText: string;
  confirmButtonText: string;
  cancelButtonText: string;
  confirmAction?: () => void;
  cancelAction?: () => void;
}

export const ModalConfirmCancel = (props: ModalConfirmCancelOwnProps) => {
  const { isOpen, headerText, bodyText, confirmButtonText, cancelButtonText, confirmAction, cancelAction } = props;
  const confirm = () => {
    typeof confirmAction === 'function' && confirmAction();
  };
  const cancel = () => {
    typeof cancelAction === 'function' && cancelAction();
  };

  return (
    <Dialog id={'modal-confirm-cancel'} open={isOpen} onClose={cancel}>
      <DialogTitle>{headerText}</DialogTitle>
      <DialogContent>
        <DialogContentText>{bodyText}</DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button id="modal-confirm-cancel-button-confirm" color="primary" onClick={confirm} autoFocus={true}>
          {confirmButtonText}
        </Button>
        <Button id="modal-confirm-cancel-button-cancel" color="secondary" onClick={cancel}>
          {cancelButtonText}
        </Button>
      </DialogActions>
    </Dialog>
  );
};
