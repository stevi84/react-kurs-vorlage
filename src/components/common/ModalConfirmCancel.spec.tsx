import { fireEvent, render, screen } from '@testing-library/react';
import { createRenderer, ShallowRenderer } from 'react-test-renderer/shallow';
import { ModalConfirmCancel } from './ModalConfirmCancel';

describe('ModalConfirmCancel', () => {
  it('should equal saved snapshot', () => {
    const renderer: ShallowRenderer = createRenderer();
    renderer.render(
      <ModalConfirmCancel
        isOpen={true}
        headerText={'header'}
        bodyText={'body'}
        confirmButtonText={'confirm'}
        cancelButtonText={'cancel'}
        confirmAction={() => {}}
        cancelAction={() => {}}
      />,
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('should execute confirm when confirmButton is pressed', () => {
    const confirmAction = jest.fn();
    const cancelAction = jest.fn();
    render(
      <ModalConfirmCancel
        isOpen={true}
        headerText={'header'}
        bodyText={'body'}
        confirmButtonText={'confirm'}
        cancelButtonText={'cancel'}
        confirmAction={confirmAction}
        cancelAction={cancelAction}
      />,
    );
    fireEvent.click(screen.getByText('confirm'));
    expect(confirmAction.mock.calls.length).toEqual(1);
    expect(cancelAction.mock.calls.length).toEqual(0);
  });

  it('should execute cancel when cancelButton is pressed', () => {
    const confirmAction = jest.fn();
    const cancelAction = jest.fn();
    render(
      <ModalConfirmCancel
        isOpen={true}
        headerText={'header'}
        bodyText={'body'}
        confirmButtonText={'confirm'}
        cancelButtonText={'cancel'}
        confirmAction={confirmAction}
        cancelAction={cancelAction}
      />,
    );
    fireEvent.click(screen.getByText('cancel'));
    expect(confirmAction.mock.calls.length).toEqual(0);
    expect(cancelAction.mock.calls.length).toEqual(1);
  });
});
