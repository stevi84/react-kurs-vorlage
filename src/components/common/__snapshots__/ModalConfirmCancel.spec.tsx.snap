// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalConfirmCancel should equal saved snapshot 1`] = `
<ForwardRef(Dialog)
  id="modal-confirm-cancel"
  onClose={[Function]}
  open={true}
>
  <ForwardRef(DialogTitle)>
    header
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <ForwardRef(DialogContentText)>
      body
    </ForwardRef(DialogContentText)>
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Button)
      autoFocus={true}
      color="primary"
      id="modal-confirm-cancel-button-confirm"
      onClick={[Function]}
    >
      confirm
    </ForwardRef(Button)>
    <ForwardRef(Button)
      color="secondary"
      id="modal-confirm-cancel-button-cancel"
      onClick={[Function]}
    >
      cancel
    </ForwardRef(Button)>
  </ForwardRef(DialogActions)>
</ForwardRef(Dialog)>
`;
